src/functions/getNextUploadBatch/framework/handler.ts

```diff
@@ -4,17 +4,25 @@ import createResponse from '../../../common/application/utils/createResponse';
 import { HttpStatus } from '../../../common/application/api/HttpStatus';
 import Response from '../../../common/application/api/Response';
 import { InterfaceTypes } from '../domain/interface-types';
-import { BatchRequestParamaterErrors } from '../domain/batch-request-param-errors';
 import { gzipSync } from 'zlib';
+import joi from '@hapi/joi';
 
 export async function handler(event: APIGatewayEvent, fnCtx: Context): Promise<Response> {
   let nextBatchData;
   const interfaceType = convertToInterfaceType(event.queryStringParameters.interface);
   const batchSize = event.queryStringParameters.batch_size;
-  const validationResponse = validateQueryParameters(interfaceType, batchSize);
 
-  if (validationResponse !== undefined) {
-    return createResponse(validationResponse, HttpStatus.BAD_REQUEST);
+  // Joi schema to handle validation of queryStringParameters
+  const batchSizeSchema = joi.object().keys({
+    interfaceType: joi.string().valid('TARS', 'RSIS', 'NOTIFY').required(),
+    batchSize: joi.number().positive().required(),
+  });
+
+  const validationResult =
+  joi.validate({ interfaceType: interfaceType.toString(), batchSize: +batchSize }, batchSizeSchema);
+
+  if (validationResult.error) {
+    return createResponse(validationResult.error, HttpStatus.BAD_REQUEST);
```

Comment:
Awesome 😍

Reply:
Is it possible that the batch size query parameter could ever be populated with something other than a number.
wonder about how it would deal with batchSize being NaN....worth a test for it? or is it highly unlikely?